master/internal/stream/system_intg_test.go

```diff
@@ -161,14 +184,39 @@ func TestStartup(t *testing.T) {
 			},
 		},
 	}
-	tester.startupMessage = &startupMessage
+	tester.StartupMessage = &startupMessage
 	err = publisherSet.entrypoint(ssupCtx, ctx, testUser, &tester, simpleUpsert)
 	require.NoError(t, err)
-	deletions, trialMsgs, err = splitDeletionsAndTrials(tester.data)
+	deletions, trialMsgs, err = splitDeletionsAndTrials(tester.Data)
 	require.NoError(t, err)
 	require.Equal(t, 1, len(deletions), "did not receive 1 deletion message")
 	require.Equal(t, "4", deletions[0], "expected deleted trials to be 4, not %s", deletions[0])
 	require.Equal(t, 0, len(trialMsgs), "received unexpected trial message")
+
+	// 3 is not known, and 4 does not exist
+	startupMessage = StartupMsg{
+		Known: KnownKeySet{
+			Trials: "1,2,4",
+		},
+		Subscribe: SubscriptionSpecSet{
+			Trials: &TrialSubscriptionSpec{
+				TrialIds: []int{1, 2, 4},
+				Since:    0,
+			},
+		},
+	}
+	tester.StartupMessage = &startupMessage
+	err = publisherSet.entrypoint(ssupCtx, ctx, testUser, &tester, simpleUpsert)
+	require.NoError(t, err)
+	deletions, trialMsgs, err = splitDeletionsAndTrials(tester.Data)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(deletions), "did not receive 1 deletion message")
+	require.Equal(t, "4", deletions[0], "expected deleted trials to be 4, not %s", deletions[0])
+	require.Equal(t, 1, len(trialMsgs), "received unexpected trial message")
+	require.Equal(t, 3, trialMsgs[0].ID, "expected trialMsg with ID 3, received ID %d",
+		trialMsgs[0].ID)
+
+	// TODO: add test that tests for diverging known key sets and subscriptions
```

Comment:
nit: the pattern we've been following for these TODO-related comments has been to use ```// XXX:```